File: zhuxiaodong/src2/main/java/com/example/demo/code/command/HttpContextUtil.java
package com.example.demo.code.command;

import lombok.val;
import lombok.var;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.HttpServletRequest;

public class HttpContextUtil {

    //获取请求对象
    public static HttpServletRequest getRequest(){
        return ((ServletRequestAttributes)RequestContextHolder.getRequestAttributes()).getRequest();
    }
    //获取ip
    public static String getIpAddress(){
        var ip="127.0.0.1";
        val request=getRequest();
        ip=request.getHeader("x-forwarder-for");
        if (ip==null||ip.length()==0||"unknown".equalsIgnoreCase(ip)){
            ip=request.getHeader("Proxy-Client-IP");
        }
        if(ip==null||ip.length()==0||"unknown".equalsIgnoreCase(ip)){
            ip=request.getHeader("WL-Client-IP");
        }
        if(ip==null||ip.length()==0||"unknown".equalsIgnoreCase(ip)){
            ip=request.getHeader("HTTP-CLIENT-IP");
        }
        if(ip==null||ip.length()==0||"unknown".equalsIgnoreCase(ip)){
            ip=request.getHeader("HTTP-FORWARDED-FOR");
        }
        if(ip==null||ip.length()==0||"unknown".equalsIgnoreCase(ip)){
            ip=request.getRemoteAddr();
        }


        System.out.println(ip);
        return "0:0:0:0:0:0:1".equals(ip) ? "127.0.0.1":ip;
    }
}

File: caiguicheng/week01/day04/上课内容/demo04/code/command/HttpContextUtil.kt
package com.example.demo04.code.command

import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes
import javax.servlet.http.HttpServletRequest

object HttpContextUtil {

    ///获取请求的对象
    @JvmStatic
    fun getRequest(): HttpServletRequest {
        return (RequestContextHolder.getRequestAttributes() as ServletRequestAttributes).request
    }

    //获取IP地址
    @JvmStatic
    fun getIpAddress(): String {
        var ip :String? = "127.0.0.1"
        val request = getRequest()
        ip = request.getHeader("x-forwarder-for")

        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("Proxy-Client-IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("WL-Proxy-Client-IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("HTTP_Client_IP")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.getHeader("HTTP_X_FORWARDED_FOR")
        }
        if (ip == null || ip.isEmpty() || "unknown".equals(ip,true)) {
            ip = request.remoteAddr
        }



        println("ip:$ip")
        return if ("0:0:0:0:0:0:0:1" === ip) "127.0.0.1" else ip
    }
}